src/main/java/gov/nist/itl/ssd/thresholding/ThresholdingProcessor.java

```diff
@@ -200,33 +215,108 @@ else if(this.thresholdingType == ThresholdingType.Yen){
 			} catch (IOException e) {
 				e.printStackTrace();
 			}*/
-	
+
 			if (threshold>-1) { 
 				//threshold it
-/*				if (doIset){
+				/*				if (doIset){
 					if (doIwhite) 
 						ip.setThreshold(threshold+1, data.length - 1, ImageProcessor.RED_LUT);//IJ.setThreshold(threshold+1, data.length - 1);
 					else
 						ip.setThreshold(0, threshold, ImageProcessor.RED_LUT);//IJ.setThreshold(0,threshold);
 				}
 				else{*/
-					for( y=0;y<ye;y++) {
-						for(x=0;x<xe;x++){
-							if(ip.getPixel(x,y)>threshold)
-								ip.putPixel(x,y,c);
-							else
-								ip.putPixel(x,y,b);
-						}
+				for( y=0;y<ye;y++) {
+					for(x=0;x<xe;x++){
+						if(ip.getPixel(x,y)>threshold)
+							ip.putPixel(x,y,c);
+						else
+							ip.putPixel(x,y,b);
 					}
-					ip.setThreshold(data.length - 1, data.length - 1, ImageProcessor.NO_LUT_UPDATE);
+				}
+				ip.setThreshold(data.length - 1, data.length - 1, ImageProcessor.NO_LUT_UPDATE);
 				//}
 			}
+
+			File outputFile = new File(outputFolder, tile.getName());
+			OMEXMLMetadata metadata = getMetadata(tile);
+			PixelType pxlType = metadata.getPixelsType(0);
+			
+			byte[] bytesArr = null;
 			
-			ImagePlus res = new ImagePlus(tile.getName(), ip);			
-			File outputImg = new File(outputFolder, tile.getName());
-			IJ.saveAsTiff(res, outputImg.getAbsolutePath());
-        }
+			switch(pxlType) {
+				case UINT8:
+					LOGGER.log(Level.INFO, tile.getName() + " is an 8bpp image");
+					bytesArr = (byte[]) ip.getPixels();
+					break;
+				case UINT16: 
+					LOGGER.log(Level.INFO, tile.getName() + " is an 16bpp image");
+					short[] shorts = (short[]) ip.getPixels();
+					
+					//Converting short array to bytes array
+					ByteBuffer byteShortBuf = ByteBuffer.allocate(shorts.length*2);
+					for (short s: shorts) {
+						byteShortBuf.putShort(s);
+					}
+					bytesArr = byteShortBuf.array();
+					break;
+				case FLOAT:
+					LOGGER.log(Level.INFO, tile.getName() + " is an 32bpp image.");
+					LOGGER.log(Level.INFO, "32bpp images are not handled by the thresholding plugin. Please convert the image to an 8bpp or a 16bpp image.");
```

Comment:
If this is not supported, log level should be warning or error, and we could throw an exception here right? Otherwise the job will finish successfully even though the resulting images are invalid

Reply:
fixed